In [1]:
# Importing libraries

import numpy as np
import pandas as pd
from pandas import ExcelWriter

In [2]:
# Reading companies data set

companies = pd.read_csv("companies.txt", sep="\t", encoding = "ISO-8859-1") # Using tab separator as the provided companies.txt file has values separated using tab

companies.head()

permalink                    name  \
0                  /Organization/-Fame                   #fame   
1               /Organization/-Qounter                :Qounter   
2  /Organization/-The-One-Of-Them-Inc-  (THE) ONE of THEM,Inc.   
3                /Organization/0-6-Com                 0-6.com   
4       /Organization/004-Technologies        004 Technologies   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
2                http://oneofthem.jp   
3                 http://www.0-6.com   
4  http://004gmbh.de/en/004-interact   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
2                                  Apps|Games|Mobile  operating          NaN   
3                                        Curated Web  operating          CHN   
4                                           Software  operating          USA   

  state_code                 region           city  founded_at  
0         16                 Mumbai         Mumbai         NaN  
1         DE             DE - Other  Delaware City  04-09-2014  
2        NaN                    NaN            NaN         NaN  
3         22                Beijing        Beijing  01-01-2007  
4         IL  Springfield, Illinois      Champaign  01-01-2010

In [3]:
# Getting general info of the data set and also to see if there are any missing values

companies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 66368 entries, 0 to 66367
Data columns (total 10 columns):
permalink        66368 non-null object
name             66367 non-null object
homepage_url     61310 non-null object
category_list    63220 non-null object
status           66368 non-null object
country_code     59410 non-null object
state_code       57821 non-null object
region           58338 non-null object
city             58340 non-null object
founded_at       51147 non-null object
dtypes: object(10)
memory usage: 5.1+ MB


In [4]:
# % of null values in each column of the companies data set

round(100*(companies.isnull().sum()/len(companies.index)), 2)

permalink         0.00
name              0.00
homepage_url      7.62
category_list     4.74
status            0.00
country_code     10.48
state_code       12.88
region           12.10
city             12.10
founded_at       22.93
dtype: float64

There seems to be a high percentage of nulls in the columns homepage_url, state_code, region, city & founded_at which we can safely ignore as we won't be using them for our analysis.

Nulls in category_list & country_code are going to be handled after rounds2 data set is merged with companies data set.

In [5]:
# Reading rounds2 data set

rounds2 = pd.read_csv("rounds2.csv", sep=",", encoding = "ISO-8859-1") # Using , separator as the provided rounds2.csv file has values separated using comma

rounds2.head()

company_permalink  \
0                  /organization/-fame   
1               /ORGANIZATION/-QOUNTER   
2               /organization/-qounter   
3  /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                /organization/0-6-com   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   

  funding_round_code   funded_at  raised_amount_usd  
0                  B  05-01-2015         10000000.0  
1                  A  14-10-2014                NaN  
2                NaN  01-03-2014           700000.0  
3                  B  30-01-2014          3406878.0  
4                  A  19-03-2008          2000000.0

In [6]:
# Getting general info of the data set and also to see if there are any missing values

rounds2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 114949 entries, 0 to 114948
Data columns (total 6 columns):
company_permalink          114949 non-null object
funding_round_permalink    114949 non-null object
funding_round_type         114949 non-null object
funding_round_code         31140 non-null object
funded_at                  114949 non-null object
raised_amount_usd          94959 non-null float64
dtypes: float64(1), object(5)
memory usage: 5.3+ MB


In [7]:
# % of null values in each column of the rounds2 data set

round(100*(rounds2.isnull().sum()/len(rounds2.index)), 2)

company_permalink           0.00
funding_round_permalink     0.00
funding_round_type          0.00
funding_round_code         72.91
funded_at                   0.00
raised_amount_usd          17.39
dtype: float64

There seems to be a high percentage of null values in funding_round_code column which we can safely ignore for now as it is not going to be used in the analysis.

We will be handling the nulls in raised_amount_usd column post merge of companies and rounds2 data sets.

# Checkpoint 1: Data Cleaning 1

Before merging the data sets, we need to make the format of permalink columns in each data set uniform so that the inner join doesn't miss out values due to case sensitivity

In [8]:
# Converting the permalink column data in both data sets to lower case to fix the duplicates

rounds2['company_permalink'] = rounds2['company_permalink'].apply(lambda x: x.lower())

companies['permalink'] = companies['permalink'].apply(lambda x: x.lower())

In [9]:
# Decoding the permalink columns data in both data sets using appropriate formats

rounds2.company_permalink = rounds2.company_permalink.str.encode('utf-8').str.decode('ascii', 'ignore')

companies.permalink = companies.permalink.str.encode('utf-8').str.decode('ascii', 'ignore')

In [10]:
# How many unique companies are present in rounds2?

len(pd.unique(rounds2['company_permalink']))

# Can also be found using => len(rounds2['company_permalink'].value_counts(dropna=False)

66368

In [11]:
# How many unique companies are present in companies?

len(pd.unique(companies['permalink']))

# Can also be found using => len(companies['permalink'].value_counts(dropna=False))

66368

In [12]:
# Are there any companies in the rounds2 file which are not present in companies?

rounds2_set = set(rounds2['company_permalink'])
companies_set = set(companies['permalink'])

rounds2_set - companies_set # Using set operations to find the difference

# Can also be found using => rounds2[rounds2['company_permalink'].apply(lambda x: x not in companies['permalink'].values)]

set()

In [13]:
# Also providing the companies that are in companies but not in rounds2

companies_set - rounds2_set # Using set operations to find the difference

set()

In [14]:
# Merging companies data set into rounds2 data set using inner join/merge

master_frame = pd.merge(rounds2, companies, how = 'inner', left_on = 'company_permalink', right_on = 'permalink')

master_frame.head()

company_permalink  \
0                  /organization/-fame   
1               /organization/-qounter   
2               /organization/-qounter   
3  /organization/-the-one-of-them-inc-   
4                /organization/0-6-com   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   

  funding_round_code   funded_at  raised_amount_usd  \
0                  B  05-01-2015         10000000.0   
1                  A  14-10-2014                NaN   
2                NaN  01-03-2014           700000.0   
3                  B  30-01-2014          3406878.0   
4                  A  19-03-2008          2000000.0   

                             permalink                    name  \
0                  /organization/-fame                   #fame   
1               /organization/-qounter                :Qounter   
2               /organization/-qounter                :Qounter   
3  /organization/-the-one-of-them-inc-  (THE) ONE of THEM,Inc.   
4                /organization/0-6-com                 0-6.com   

             homepage_url                                      category_list  \
0      http://livfame.com                                              Media   
1  http://www.qounter.com  Application Platforms|Real Time|Social Network...   
2  http://www.qounter.com  Application Platforms|Real Time|Social Network...   
3     http://oneofthem.jp                                  Apps|Games|Mobile   
4      http://www.0-6.com                                        Curated Web   

      status country_code state_code      region           city  founded_at  
0  operating          IND         16      Mumbai         Mumbai         NaN  
1  operating          USA         DE  DE - Other  Delaware City  04-09-2014  
2  operating          USA         DE  DE - Other  Delaware City  04-09-2014  
3  operating          NaN        NaN         NaN            NaN         NaN  
4  operating          CHN         22     Beijing        Beijing  01-01-2007

In [15]:
# How many observations are present in master_frame?

master_frame.info() # Also to find the general info and null values

<class 'pandas.core.frame.DataFrame'>
Int64Index: 114949 entries, 0 to 114948
Data columns (total 16 columns):
company_permalink          114949 non-null object
funding_round_permalink    114949 non-null object
funding_round_type         114949 non-null object
funding_round_code         31140 non-null object
funded_at                  114949 non-null object
raised_amount_usd          94959 non-null float64
permalink                  114949 non-null object
name                       114948 non-null object
homepage_url               108815 non-null object
category_list              111539 non-null object
status                     114949 non-null object
country_code               106271 non-null object
state_code                 104003 non-null object
region                     104782 non-null object
city                       104785 non-null object
founded_at                 94428 non-null object
dtypes: float64(1), object(15)
memory usage: 14.9+ MB


We can proceed with dropping the columns that are not needed for the analysis from the master_frame data set

In [16]:
# Drop funding_round_permalink, permalink, funding_round_code, funded_at, homepage_url, state_code, region, city, founded_at columns from master_frame

drop_col = ['funding_round_permalink','permalink','funding_round_code','funded_at','homepage_url','state_code','region','city','founded_at']

master_frame.drop(drop_col, axis=1, inplace=True)

master_frame.isnull().sum(axis=0) # Find the sum of missing values in each column in master_frame data set

company_permalink         0
funding_round_type        0
raised_amount_usd     19990
name                      1
category_list          3410
status                    0
country_code           8678
dtype: int64

In [17]:
# % of null values in each column of the master_frame data set

round(100*(master_frame.isnull().sum()/len(master_frame.index)), 2)

company_permalink      0.00
funding_round_type     0.00
raised_amount_usd     17.39
name                   0.00
category_list          2.97
status                 0.00
country_code           7.55
dtype: float64

Since raised_amount_usd, name, category_list & country_code are each critical for the analysis, we can't have them as nulls.

Now, null values of name, category_list & country_code can't be imputed using statistical measures as they are not numeric attributes but also can't be assumed to have some value among the existing ones. The logical way to handle them is to delete the rows with null values.

Now, null values of raised_amount_usd column can't be imputed using statistical measures as it would be inaccurate since it depends on multiple attributes like category, country etc. The logical way to handle them is to delete the rows with null values.

In [18]:
# Drop null values of raised_amount_usd, name, category_list, country_code

master_frame = master_frame[pd.notnull(master_frame['raised_amount_usd'])]
master_frame = master_frame[pd.notnull(master_frame['name'])]
master_frame = master_frame[pd.notnull(master_frame['category_list'])]
master_frame = master_frame[pd.notnull(master_frame['country_code'])]

In [19]:
# % of null values in each column of the master_frame data set

round(100*(master_frame.isnull().sum()/len(master_frame.index)), 2)

company_permalink     0.0
funding_round_type    0.0
raised_amount_usd     0.0
name                  0.0
category_list         0.0
status                0.0
country_code          0.0
dtype: float64

In [20]:
master_frame.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 88528 entries, 0 to 114947
Data columns (total 7 columns):
company_permalink     88528 non-null object
funding_round_type    88528 non-null object
raised_amount_usd     88528 non-null float64
name                  88528 non-null object
category_list         88528 non-null object
status                88528 non-null object
country_code          88528 non-null object
dtypes: float64(1), object(6)
memory usage: 5.4+ MB


In [21]:
'''
writer = ExcelWriter('/Users/sdittaka/Downloads/master_frame.xlsx', engine='xlsxwriter', options={'strings_to_urls': False})
master_frame.to_excel(writer,'Sheet1')
writer.save()
'''

# Checkpoint 2: Funding Type Analysis

In [22]:
# Number of values for each funding round type

master_frame.funding_round_type.value_counts()

venture                  47809
seed                     21094
debt_financing            6506
angel                     4400
grant                     1939
private_equity            1820
undisclosed               1345
convertible_note          1320
equity_crowdfunding       1128
post_ipo_equity            598
product_crowdfunding       330
post_ipo_debt              151
non_equity_assistance       60
secondary_market            28
Name: funding_round_type, dtype: int64

In [23]:
# Obtaining average funding amount for the interested funding types

print("Venture type average funding amount is {} Million USD ".format(str(master_frame.loc[master_frame.funding_round_type == 'venture'].raised_amount_usd.mean()/1000000)))
print("Angel type average funding amount is {} Million USD ".format(str(master_frame.loc[master_frame.funding_round_type == 'angel'].raised_amount_usd.mean()/1000000)))
print("Seed type average funding amount is {} Million USD ".format(str(master_frame.loc[master_frame.funding_round_type == 'seed'].raised_amount_usd.mean()/1000000)))
print("Private Equity type average funding amount is {} Million USD ".format(str(master_frame.loc[master_frame.funding_round_type == 'private_equity'].raised_amount_usd.mean()/1000000)))

Venture type average funding amount is 11.724222690204773 Million USD 
Angel type average funding amount is 0.9715738911363636 Million USD 
Seed type average funding amount is 0.7478279478524699 Million USD 
Private Equity type average funding amount is 73.93848627527473 Million USD 


Since, Spark Funds wants to invest between 5 to 15 million USD per investment round, "venture" is the funding type that is most suitable for it's investment as it's average 11.72 Million USD falls in this range

# Checkpoint 3: Country Analysis

In [24]:
# Obtaining all the observations for the funding type 'venture'

venture_frame = master_frame.loc[master_frame.funding_round_type == 'venture']

venture_frame.head()

company_permalink funding_round_type  raised_amount_usd  \
0                   /organization/-fame            venture         10000000.0   
4                 /organization/0-6-com            venture          2000000.0   
8   /organization/0ndine-biomedical-inc            venture           719491.0   
10                 /organization/0xdata            venture         20000000.0   
11                 /organization/0xdata            venture          1700000.0   

                      name  category_list     status country_code  
0                    #fame          Media  operating          IND  
4                  0-6.com    Curated Web  operating          CHN  
8   Ondine Biomedical Inc.  Biotechnology  operating          CAN  
10                  H2O.ai      Analytics  operating          USA  
11                  H2O.ai      Analytics  operating          USA

In [28]:
# Using pivot table on the venture data frame using sum aggregate function to obtain the top 9 countries based on the total investment amount each country received

top9 = venture_frame.pivot_table(values = 'raised_amount_usd', index = 'country_code', aggfunc = 'sum').sort_values(by='raised_amount_usd', ascending = False)[0:9]

top9.raised_amount_usd = top9.raised_amount_usd/1000000

top9.rename(columns={"raised_amount_usd":"raised_amount_usd (in millions)"}, inplace=True)

top9

raised_amount_usd (in millions)
country_code                                 
USA                             420068.029342
CHN                              39338.918773
GBR                              20072.813004
IND                              14261.508718
CAN                               9482.217668
FRA                               7226.851352
ISR                               6854.350477
DEU                               6306.921981
JPN                               3167.647127

From http://www.emmir.org/fileadmin/user_upload/admission/Countries_where_English_is_an_official_language.pdf we can confirm that USA, GBR & IND are the Top 3 English-Speaking Countries

In [26]:
writer = ExcelWriter('/Users/sdittaka/Downloads/top9.xlsx', engine='xlsxwriter', options={'strings_to_urls': False})
top9.to_excel(writer,'Sheet1')
writer.save()

# Checkpoint 4: Sector Analysis 1

In [29]:
# Reading mapping data set

mapping = pd.read_csv("mapping.csv", sep=",", encoding = "ISO-8859-1") # Using , separator as the provided companies.txt file has values separated using comma

mapping.head()

category_list  Automotive & Sports  Blanks  Cleantech / Semiconductors  \
0            NaN                    0       1                           0   
1             3D                    0       0                           0   
2    3D Printing                    0       0                           0   
3  3D Technology                    0       0                           0   
4     Accounting                    0       0                           0   

   Entertainment  Health  Manufacturing  News, Search and Messaging  Others  \
0              0       0              0                           0       0   
1              0       0              1                           0       0   
2              0       0              1                           0       0   
3              0       0              1                           0       0   
4              0       0              0                           0       0   

   Social, Finance, Analytics, Advertising  
0                                        0  
1                                        0  
2                                        0  
3                                        0  
4                                        1

In [30]:
# Getting general info of the data set and also to see if there are any missing values

mapping.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 688 entries, 0 to 687
Data columns (total 10 columns):
category_list                              687 non-null object
Automotive & Sports                        688 non-null int64
Blanks                                     688 non-null int64
Cleantech / Semiconductors                 688 non-null int64
Entertainment                              688 non-null int64
Health                                     688 non-null int64
Manufacturing                              688 non-null int64
News, Search and Messaging                 688 non-null int64
Others                                     688 non-null int64
Social, Finance, Analytics, Advertising    688 non-null int64
dtypes: int64(9), object(1)
memory usage: 53.8+ KB


There is one missing value (which happens to be the first row in the data set) in the category_list column and that corresponds to the 'Blanks' column. Since, this row and 'Blanks' column are not useful for the analysis, proceeding with dropping them from the data set.

In [31]:
mapping = mapping.drop(columns=['Blanks']) # Drop 'Blanks' column
mapping = mapping.loc[1:687,:] # Skipping the first row

mapping.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 687 entries, 1 to 687
Data columns (total 9 columns):
category_list                              687 non-null object
Automotive & Sports                        687 non-null int64
Cleantech / Semiconductors                 687 non-null int64
Entertainment                              687 non-null int64
Health                                     687 non-null int64
Manufacturing                              687 non-null int64
News, Search and Messaging                 687 non-null int64
Others                                     687 non-null int64
Social, Finance, Analytics, Advertising    687 non-null int64
dtypes: int64(8), object(1)
memory usage: 48.4+ KB


In [32]:
# Examining the data frame

mapping.head()

category_list  Automotive & Sports  Cleantech / Semiconductors  \
1                3D                    0                           0   
2       3D Printing                    0                           0   
3     3D Technology                    0                           0   
4        Accounting                    0                           0   
5  Active Lifestyle                    0                           0   

   Entertainment  Health  Manufacturing  News, Search and Messaging  Others  \
1              0       0              1                           0       0   
2              0       0              1                           0       0   
3              0       0              1                           0       0   
4              0       0              0                           0       0   
5              0       1              0                           0       0   

   Social, Finance, Analytics, Advertising  
1                                        0  
2                                        0  
3                                        0  
4                                        1  
5                                        0

In [33]:
# Obtaining the main_sector for each row where there is a value 1 into a new data frame with the corresponding category

mapping_new = pd.DataFrame(index=range(len(mapping)), columns=['category_list','main_sector'])

for index, row in mapping.iterrows():
    mapping_new.loc[index] = [row['category_list'], str(row[row == 1].index[0])]

mapping_new.head()

category_list                              main_sector
0            NaN                                      NaN
1             3D                            Manufacturing
2    3D Printing                            Manufacturing
3  3D Technology                            Manufacturing
4     Accounting  Social, Finance, Analytics, Advertising

In [34]:
mapping_new = mapping_new.loc[1:687,:] # To get rid of the first NaN row

mapping_new.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 687 entries, 1 to 687
Data columns (total 2 columns):
category_list    687 non-null object
main_sector      687 non-null object
dtypes: object(2)
memory usage: 16.1+ KB


By carefully examining the raw mapping data set, we can see that the strings "na" and "Na" has been replaced with value 0 which resulted in values like "Analytics" shown as "A0lytics".

To handle this, we need to replace the occurences of 0 only in the string portion of the category_list values with "na" string.

To make it uniform, we convert the category_list to lower case values.

In [35]:
mapping_new['category_list'] = mapping_new['category_list'].apply(lambda x : x.lower())

mapping_new['category_list'] = mapping_new['category_list'].apply(lambda x : x.replace("0","na") if not ".0" in x else x)

mapping_new.head()

category_list                              main_sector
1                3d                            Manufacturing
2       3d printing                            Manufacturing
3     3d technology                            Manufacturing
4        accounting  Social, Finance, Analytics, Advertising
5  active lifestyle                                   Health

In [36]:
master_frame.category_list = master_frame.category_list.astype('str') # Converting the data type to string to perform string operations

master_frame['primary_sector'] = master_frame.category_list.apply(lambda x: x.split('|')[0]) # Extracting the primary sector of each category list from the category_list column

master_frame['primary_sector'] = master_frame['primary_sector'].apply(lambda x: x.lower()) # Converting to lower case for uniformity

master_frame.head()

company_permalink funding_round_type  raised_amount_usd  \
0                  /organization/-fame            venture         10000000.0   
2               /organization/-qounter               seed           700000.0   
4                /organization/0-6-com            venture          2000000.0   
6     /organization/01games-technology        undisclosed            41250.0   
7  /organization/0ndine-biomedical-inc               seed            43360.0   

                     name                                      category_list  \
0                   #fame                                              Media   
2                :Qounter  Application Platforms|Real Time|Social Network...   
4                 0-6.com                                        Curated Web   
6      01Games Technology                                              Games   
7  Ondine Biomedical Inc.                                      Biotechnology   

      status country_code         primary_sector  
0  operating          IND                  media  
2  operating          USA  application platforms  
4  operating          CHN            curated web  
6  operating          HKG                  games  
7  operating          CAN          biotechnology

In [37]:
# Finding the categories that are in master frame but not in mapping as these observations are not going to be part of our sector analysis

master_frame_set = set(master_frame['primary_sector'])
mapping_set = set(mapping_new['category_list'])

master_frame_set - mapping_set

{'adaptive equipment',
 'biotechnology and semiconductor',
 'cause marketing',
 'deep information technology',
 'english-speaking',
 'enterprise hardware',
 'generation y-z',
 'golf equipment',
 'google glass',
 'greentech',
 'group email',
 'internet technology',
 'internet tv',
 'kinect',
 'lingerie',
 'mobile emergency&health',
 'natural gas uses',
 'nightlife',
 'product search',
 'psychology',
 'racing',
 'rapidly expanding',
 'real estate investors',
 'registrars',
 'retirement',
 'sex industry',
 'skill gaming',
 'social media advertising',
 'spas',
 'specialty retail',
 'sponsorship',
 'subscription businesses',
 'swimming',
 'toys',
 'vacation rentals'}

In [38]:
# Finding the categories that are in mapping frame but not in master_frame as these observations are not going to be part of our sector analysis

mapping_set - master_frame_set

{'energy storage',
 'film distribution',
 'flowers',
 'lotteries',
 'low bid auctions',
 'multi-level marketing',
 'pervasive computing',
 'plumbers',
 'smart grid',
 'social recruiting',
 'taxis',
 'veterinary',
 'virtual workforces',
 'women'}

In [39]:
# Examining the number of observations that are going to be missed as part of the merge

merged_frame = pd.merge(master_frame, mapping_new, how = 'left', left_on = 'primary_sector', right_on = 'category_list')

merged_frame.isnull().sum(axis=0)

company_permalink      0
funding_round_type     0
raised_amount_usd      0
name                   0
category_list_x        0
status                 0
country_code           0
primary_sector         0
category_list_y       63
main_sector           63
dtype: int64

In [40]:
# Examining the observations that are going to be missed as part of the merge

merged_frame.loc[merged_frame.main_sector.isnull()].loc[:,['primary_sector','main_sector']].sort_values(by='primary_sector').drop_duplicates()

primary_sector main_sector
64733               adaptive equipment         NaN
33669  biotechnology and semiconductor         NaN
1995                   cause marketing         NaN
85941      deep information technology         NaN
6358                  english-speaking         NaN
13745              enterprise hardware         NaN
10772                   generation y-z         NaN
85061                   golf equipment         NaN
34904                     google glass         NaN
67761                        greentech         NaN
41213                      group email         NaN
16218              internet technology         NaN
24709                      internet tv         NaN
58939                           kinect         NaN
71679                         lingerie         NaN
81049          mobile emergency&health         NaN
63748                 natural gas uses         NaN
17156                        nightlife         NaN
54109                   product search         NaN
61263                       psychology         NaN
27532                           racing         NaN
40079                rapidly expanding         NaN
4947             real estate investors         NaN
30684                       registrars         NaN
68301                       retirement         NaN
59611                     sex industry         NaN
25706                     skill gaming         NaN
13752         social media advertising         NaN
54032                             spas         NaN
28076                 specialty retail         NaN
81778                      sponsorship         NaN
69512          subscription businesses         NaN
28945                         swimming         NaN
9760                              toys         NaN
34424                 vacation rentals         NaN

In [41]:
# Merge master_frame with mapping_new on sector using inner join

master_frame = pd.merge(master_frame, mapping_new, how = 'inner', left_on = 'primary_sector', right_on = 'category_list')
master_frame.drop('category_list_y', axis=1, inplace=True) # Dropping the duplicate category list column
master_frame.rename(columns={"category_list_x":"category_list"}, inplace=True) # Rename to appropriate name
master_frame.head()

company_permalink   funding_round_type  raised_amount_usd  \
0           /organization/-fame              venture         10000000.0   
1           /organization/90min              venture         15000000.0   
2           /organization/90min              venture          5800000.0   
3           /organization/90min              venture         18000000.0   
4  /organization/a-dance-for-me  equity_crowdfunding          1090000.0   

             name                        category_list     status  \
0           #fame                                Media  operating   
1           90min  Media|News|Publishing|Soccer|Sports  operating   
2           90min  Media|News|Publishing|Soccer|Sports  operating   
3           90min  Media|News|Publishing|Soccer|Sports  operating   
4  A Dance for Me       Media|News|Photo Sharing|Video  operating   

  country_code primary_sector    main_sector  
0          IND          media  Entertainment  
1          GBR          media  Entertainment  
2          GBR          media  Entertainment  
3          GBR          media  Entertainment  
4          USA          media  Entertainment

In [42]:
# Examining the resultant data frame for number of observations

master_frame.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 88465 entries, 0 to 88464
Data columns (total 9 columns):
company_permalink     88465 non-null object
funding_round_type    88465 non-null object
raised_amount_usd     88465 non-null float64
name                  88465 non-null object
category_list         88465 non-null object
status                88465 non-null object
country_code          88465 non-null object
primary_sector        88465 non-null object
main_sector           88465 non-null object
dtypes: float64(1), object(8)
memory usage: 6.7+ MB


# Checkpoint 5: Sector Analysis 2

In [43]:
'''
writer = ExcelWriter('/Users/sdittaka/Downloads/master_frame.xlsx', engine='xlsxwriter', options={'strings_to_urls': False})
master_frame.to_excel(writer,'Sheet1')
writer.save()
'''

In [44]:
# Creating D1 frame for the observations of USA, funding type venture and the raised amount within the 5-15 million USD range

D1 = master_frame.loc[(master_frame['country_code'] == 'USA') & (master_frame['funding_round_type'] == 'venture') & (master_frame['raised_amount_usd'] >= 5000000) & (master_frame['raised_amount_usd'] <= 15000000)]

# The total number (or count) of investments for each main sector in a separate column

D1['main_sector_count'] = D1['main_sector'].apply(lambda x: D1.main_sector.value_counts()[x])

# The total amount invested in each main sector in a separate column

D1['main_sector_investment'] = D1['main_sector'].apply(lambda x: D1.groupby('main_sector')['raised_amount_usd'].sum()[x])

# Examining the data frame

D1.head()

/Applications/anaconda/lib/python3.6/site-packages/ipykernel/__main__.py:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
/Applications/anaconda/lib/python3.6/site-packages/ipykernel/__main__.py:11: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy


company_permalink funding_round_type  raised_amount_usd  \
7   /organization/all-def-digital            venture          5000000.0   
31       /organization/chefs-feed            venture          5000000.0   
61   /organization/huffingtonpost            venture          5000000.0   
62   /organization/huffingtonpost            venture          5000000.0   
85        /organization/matchmine            venture         10000000.0   

                   name                           category_list     status  \
7       All Def Digital                                   Media  operating   
31            ChefsFeed     Media|Mobile|Restaurants|Technology  operating   
61  The Huffington Post                   Media|News|Publishing   acquired   
62  The Huffington Post                   Media|News|Publishing   acquired   
85            MatchMine  Media|News|Reviews and Recommendations     closed   

   country_code primary_sector    main_sector  main_sector_count  \
7           USA          media  Entertainment                591   
31          USA          media  Entertainment                591   
61          USA          media  Entertainment                591   
62          USA          media  Entertainment                591   
85          USA          media  Entertainment                591   

    main_sector_investment  
7             5.099198e+09  
31            5.099198e+09  
61            5.099198e+09  
62            5.099198e+09  
85            5.099198e+09

In [45]:
# Creating D2 frame for the observations of GBR, funding type venture and the raised amount within the 5-15 million USD range

D2 = master_frame.loc[(master_frame['country_code'] == 'GBR') & (master_frame['funding_round_type'] == 'venture') & (master_frame['raised_amount_usd'] >= 5000000) & (master_frame['raised_amount_usd'] <= 15000000)]

# The total number (or count) of investments for each main sector in a separate column

D2['main_sector_count'] = D2['main_sector'].apply(lambda x: D2.main_sector.value_counts()[x])

# The total amount invested in each main sector in a separate column

D2['main_sector_investment'] = D2['main_sector'].apply(lambda x: D2.groupby('main_sector')['raised_amount_usd'].sum()[x])

# Examining the data frame

D2.head()

/Applications/anaconda/lib/python3.6/site-packages/ipykernel/__main__.py:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
/Applications/anaconda/lib/python3.6/site-packages/ipykernel/__main__.py:11: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy


company_permalink funding_round_type  \
1                          /organization/90min            venture   
2                          /organization/90min            venture   
225  /organization/common-interest-communities            venture   
257                /organization/geospock-ltd-            venture   
258                /organization/geospock-ltd-            venture   

     raised_amount_usd                         name  \
1           15000000.0                        90min   
2            5800000.0                        90min   
225         10000000.0  Common Interest Communities   
257          5460000.0                GeoSpock Ltd.   
258          5400000.0                GeoSpock Ltd.   

                                        category_list     status country_code  \
1                 Media|News|Publishing|Soccer|Sports  operating          GBR   
2                 Media|News|Publishing|Soccer|Sports  operating          GBR   
225  Application Platforms|Internet|Software|Startups  operating          GBR   
257         Application Platforms|Databases|Real Time  operating          GBR   
258         Application Platforms|Databases|Real Time  operating          GBR   

            primary_sector                 main_sector  main_sector_count  \
1                    media               Entertainment                 56   
2                    media               Entertainment                 56   
225  application platforms  News, Search and Messaging                 73   
257  application platforms  News, Search and Messaging                 73   
258  application platforms  News, Search and Messaging                 73   

     main_sector_investment  
1               482784687.0  
2               482784687.0  
225             615746235.0  
257             615746235.0  
258             615746235.0

In [46]:
# Creating D2 frame for the observations of GBR, funding type venture and the raised amount within the 5-15 million USD range

D3 = master_frame.loc[(master_frame['country_code'] == 'IND') & (master_frame['funding_round_type'] == 'venture') & (master_frame['raised_amount_usd'] >= 5000000) & (master_frame['raised_amount_usd'] <= 15000000)]

# The total number (or count) of investments for each main sector in a separate column

D3['main_sector_count'] = D3['main_sector'].apply(lambda x: D3.main_sector.value_counts()[x])

# The total amount invested in each main sector in a separate column

D3['main_sector_investment'] = D3['main_sector'].apply(lambda x: D3.groupby('main_sector')['raised_amount_usd'].sum()[x])

# Examining the data frame

D3.head()

/Applications/anaconda/lib/python3.6/site-packages/ipykernel/__main__.py:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
/Applications/anaconda/lib/python3.6/site-packages/ipykernel/__main__.py:11: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy


company_permalink funding_round_type  raised_amount_usd  \
0               /organization/-fame            venture         10000000.0   
550           /organization/babajob            venture         10000000.0   
593  /organization/bharat-matrimony            venture         11750000.0   
640     /organization/bluestone-com            venture          5000000.0   
642     /organization/bluestone-com            venture         10000000.0   

                 name                                      category_list  \
0               #fame                                              Media   
550           Babajob  Curated Web|Information Technology|Services|St...   
593  Bharat Matrimony                           Curated Web|Match-Making   
640     Bluestone.com                                        Curated Web   
642     Bluestone.com                                        Curated Web   

        status country_code primary_sector                 main_sector  \
0    operating          IND          media               Entertainment   
550  operating          IND    curated web  News, Search and Messaging   
593  operating          IND    curated web  News, Search and Messaging   
640  operating          IND    curated web  News, Search and Messaging   
642  operating          IND    curated web  News, Search and Messaging   

     main_sector_count  main_sector_investment  
0                   33             280830000.0  
550                 52             433834545.0  
593                 52             433834545.0  
640                 52             433834545.0  
642                 52             433834545.0

In [ ]:
'''
D1_Writer = ExcelWriter('/Users/sdittaka/Downloads/D1.xlsx', engine='xlsxwriter', options={'strings_to_urls': False})
D1.to_excel(D1_Writer,'Sheet1')
D1_Writer.save()

D2_Writer = ExcelWriter('/Users/sdittaka/Downloads/D2.xlsx', engine='xlsxwriter', options={'strings_to_urls': False})
D2.to_excel(D2_Writer,'Sheet1')
D2_Writer.save()

D3_Writer = ExcelWriter('/Users/sdittaka/Downloads/D3.xlsx', engine='xlsxwriter', options={'strings_to_urls': False})
D3.to_excel(D3_Writer,'Sheet1')
D3_Writer.save()
'''

### Total number of Investments (count)

In [47]:
D1.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 12063 entries, 7 to 88457
Data columns (total 11 columns):
company_permalink         12063 non-null object
funding_round_type        12063 non-null object
raised_amount_usd         12063 non-null float64
name                      12063 non-null object
category_list             12063 non-null object
status                    12063 non-null object
country_code              12063 non-null object
primary_sector            12063 non-null object
main_sector               12063 non-null object
main_sector_count         12063 non-null int64
main_sector_investment    12063 non-null float64
dtypes: float64(2), int64(1), object(8)
memory usage: 1.1+ MB


In [48]:
D2.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 621 entries, 1 to 88265
Data columns (total 11 columns):
company_permalink         621 non-null object
funding_round_type        621 non-null object
raised_amount_usd         621 non-null float64
name                      621 non-null object
category_list             621 non-null object
status                    621 non-null object
country_code              621 non-null object
primary_sector            621 non-null object
main_sector               621 non-null object
main_sector_count         621 non-null int64
main_sector_investment    621 non-null float64
dtypes: float64(2), int64(1), object(8)
memory usage: 58.2+ KB


In [49]:
D3.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 328 entries, 0 to 88157
Data columns (total 11 columns):
company_permalink         328 non-null object
funding_round_type        328 non-null object
raised_amount_usd         328 non-null float64
name                      328 non-null object
category_list             328 non-null object
status                    328 non-null object
country_code              328 non-null object
primary_sector            328 non-null object
main_sector               328 non-null object
main_sector_count         328 non-null int64
main_sector_investment    328 non-null float64
dtypes: float64(2), int64(1), object(8)
memory usage: 30.8+ KB


### Total amount of investment (USD)

In [50]:
print('Total amount of investment (USD) for USA is {} Million USD'.format(str(D1.raised_amount_usd.sum()/1000000)))

Total amount of investment (USD) for USA is 107757.097294 Million USD


In [51]:
print('Total amount of investment (USD) for GBR is {} Million USD'.format(str(D2.raised_amount_usd.sum()/1000000)))

Total amount of investment (USD) for GBR is 5379.078691 Million USD


In [52]:
print('Total amount of investment (USD) for IND is {} Million USD'.format(str(D3.raised_amount_usd.sum()/1000000)))

Total amount of investment (USD) for IND is 2949.543602 Million USD


### Top/Second/Third Sector name & no. of investments

In [53]:
# For USA, top 3 sectors (based on count of investments) and the corresponding count

D1.loc[:,['main_sector','main_sector_count']].sort_values(by='main_sector_count', ascending=False).drop_duplicates(keep='first', inplace=False)[0:3]

# Can also be obtained using => D1['main_sector'].value_counts()

main_sector  main_sector_count
88457                                   Others               2950
52435  Social, Finance, Analytics, Advertising               2714
75724               Cleantech / Semiconductors               2350

In [54]:
# For GBR, top 3 sectors (based on count of investments) and the corresponding count

D2.loc[:,['main_sector','main_sector_count']].sort_values(by='main_sector_count', ascending=False).drop_duplicates(keep='first', inplace=False)[0:3]

# Can also be obtained using => D2['main_sector'].value_counts()

main_sector  main_sector_count
38526                                   Others                147
44688  Social, Finance, Analytics, Advertising                133
50725               Cleantech / Semiconductors                130

In [55]:
# For IND, top 3 sectors (based on count of investments) and the corresponding count

D3.loc[:,['main_sector','main_sector_count']].sort_values(by='main_sector_count', ascending=False).drop_duplicates(keep='first', inplace=False)[0:3]

# Can also be obtained using => D3['main_sector'].value_counts()

main_sector  main_sector_count
67060                                   Others                110
48445  Social, Finance, Analytics, Advertising                 60
2291                News, Search and Messaging                 52

### For point 3 (top sector count-wise), which company received the highest investment?

In [56]:
# For USA, "Virtustream" is the company which received the highest investment in "Others" sector

D1_others_df = D1.loc[D1.main_sector == 'Others']

D1_others_df.pivot_table(values = 'raised_amount_usd', index = 'name', aggfunc = 'sum').sort_values(by='raised_amount_usd', ascending = False)

raised_amount_usd
name                                             
Virtustream                            64300000.0
Capella Photonics                      54968051.0
AirTight Networks                      54201907.0
deCarta                                52100000.0
Black Duck Software                    51000000.0
Approva                                50100000.0
Five9                                  49600000.0
Bit9                                   48433533.0
Aryaka Networks                        48166500.0
BTI Systems                            48000000.0
Change Healthcare                      47000000.0
Zilliant                               45600000.0
Vormetric                              45000000.0
Redfin                                 44800000.0
Elemental Technologies                 44100000.0
Blazent                                44075260.0
3VR                                    44000000.0
Arena Solutions                        42797302.0
Convio                                 42750000.0
Taulia                                 42500000.0
Colubris Networks                      42000000.0
StoredIQ                               41836657.0
Appcelerator                           41800000.0
ExaGrid Systems                        41717024.0
RF Code                                41540000.0
Delivery Agent                         41447137.0
MOGL                                   41436966.0
ForeScout Technologies, Inc.           41250000.0
Tillster                               41198945.0
ScienceLogic                           41000000.0
...                                           ...
CX                                      5000000.0
Yapta                                   5000000.0
Predictive Science                      5000000.0
Previstar                               5000000.0
Fooda                                   5000000.0
CLK Design Automation                   5000000.0
Frederick's of Hollywood Group          5000000.0
NuView Systems                          5000000.0
Fuego                                   5000000.0
Pet's Choice                            5000000.0
Caringo                                 5000000.0
Carlotz                                 5000000.0
Workspace                               5000000.0
SlideRocket                             5000000.0
Fashion Playtes                         5000000.0
Ciespace                                5000000.0
SynapticMash                            5000000.0
FlatBurger                              5000000.0
Focus Automated Equities                5000000.0
Food Runners                            5000000.0
Peak8 Partners                          5000000.0
Peek Kids                               5000000.0
Catavolt                                5000000.0
PeakStream                              5000000.0
X-1                                     5000000.0
CashWorks                               5000000.0
Peatix                                  5000000.0
UrbanBound                              5000000.0
XMPie                                   5000000.0
Joinem                                  5000000.0

[2049 rows x 1 columns]

In [57]:
# For GBR, "Electric Cloud" is the company which received the highest investment in "Others" sector

D2_others_df = D2.loc[D2.main_sector == 'Others']

D2_others_df.pivot_table(values = 'raised_amount_usd', index = 'name', aggfunc = 'sum').sort_values(by='raised_amount_usd', ascending = False)

raised_amount_usd
name                                           
Electric Cloud                       37000000.0
SenSage                              36250000.0
Enigmatec                            32500000.0
SilverRail Technologies              29000000.0
OpenCloud                            27972766.0
MyOptique Group                      27694613.0
Notonthehighstreet                   26900000.0
ACS Clothing                         26239014.0
Tribold                              26000000.0
ip.access                            25000000.0
Eglue Business Technologies          22000000.0
Alfresco                             21000000.0
Aepona                               20000000.0
Lyst                                 19000000.0
Njini                                18000000.0
Cambridge Broadband Networks         16500000.0
EnvironmentIQ                        15000000.0
Azure Solutions                      15000000.0
KaliteUKASH                          14400000.0
Ocapo                                14394888.0
Celona Technologies                  14000000.0
State                                14000000.0
Helveta                              13697597.0
BathEmpire                           12879637.0
Six Degrees Group                    12529381.0
Fits.me                              12500000.0
Prodigy Finance                      12500000.0
onefinestay                          12200000.0
Bathrooms.com                        12060066.0
Velocity                             12000000.0
...                                         ...
NewVoiceMedia                         6300000.0
Rhetorical Systems                    6282049.0
Marcandi                              6190000.0
MadBid.com                            6161968.0
Bluebell Telecom                      6124490.0
Usentric                              6110000.0
Blackbay                              6110000.0
deltaDOT                              6100000.0
TSSI Systems                          6084330.0
eCourier.co.uk                        6064606.0
Barburrito                            6022987.0
Origami Energy                        5809932.0
Trampoline Systems                    5799119.0
Rootstock Software                    5750000.0
Flexiant                              5716641.0
SiteMinder                            5715957.0
Deep-Secure                           5596159.0
Biometric Security                    5500000.0
Clarisite                             5500000.0
ZBD Displays                          5430000.0
Almondy                               5407412.0
Gear4music.com                        5385352.0
Medikidz                              5055753.0
ParcelGenie                           5000000.0
Udex                                  5000000.0
Bacula Systems                        5000000.0
Vector Watch                          5000000.0
The Filter                            5000000.0
BullGuard                             5000000.0
PhotoBox                              5000000.0

[120 rows x 1 columns]

In [58]:
# For IND, "FirstCry.com" is the company which received the highest investment in "Others" sector

D3_others_df = D3.loc[D3.main_sector == 'Others']

D3_others_df.pivot_table(values = 'raised_amount_usd', index = 'name', aggfunc = 'sum').sort_values(by='raised_amount_usd', ascending = False)

raised_amount_usd
name                                                                 
FirstCry.com                                               39000000.0
Myntra                                                     38000000.0
CommonFloor                                                32900000.0
Pepperfry.com                                              28000000.0
ItzCash Card Ltd.                                          25000000.0
NxtGen Data Center & Cloud Services                        22300000.0
Maharana Infrastructure and Professional Servic...         21600000.0
Comat Technologies                                         21000000.0
CaratLane                                                  21000000.0
Roposo                                                     20000000.0
LimeRoad                                                   20000000.0
India Property Online                                      19000000.0
Fashion & You                                              18000000.0
Hippocampus Learning Centres                               18000000.0
ZO Rooms                                                   15000000.0
GirnarSoft                                                 15000000.0
Voylla Retail Pvt. Ltd.                                    15000000.0
Tejas Networks India                                       15000000.0
K-12 Techno Services                                       15000000.0
Classteacher Learning Systems                              15000000.0
iProf Learning Solutions                                   15000000.0
iPolicy Networks                                           15000000.0
HomeShop18                                                 14000000.0
PayMate India                                              14000000.0
EximSoft-Trianz                                            14000000.0
Quick Heal Technologies                                    13000000.0
Power2SME                                                  12800000.0
Prabhat fresh                                              12800000.0
Firepro Systems                                            12000000.0
Ariisto                                                    12000000.0
...                                                               ...
eYantra Industries                                          7500000.0
Sapience Analytics Private Limited                          7400000.0
Opinio                                                      7000000.0
CarWale                                                     7000000.0
Ittiam Systems (Pvt)                                        6500000.0
Canvera Digital Technologies                                6500000.0
21Diamonds                                                  6369507.0
mydala                                                      6000000.0
Teabox                                                      6000000.0
Treebo Hotels                                               6000000.0
Meditrina Hospital                                          6000000.0
ConnectM Technology Solutions                               6000000.0
Conquest                                                    6000000.0
Delhivery                                                   6000000.0
Citrus Payment                                              5500000.0
Monarch Innovative Technologies                             5500000.0
GoldVIP Technology Solutions (Crown-it)                     5500000.0
Aujas Networks                                              5400000.0
Hurix Systems Private                                       5100000.0
Vriti Infocom                                               5000000.0
Appsdaily Solutions                                         5000000.0
Securens                                                    5000000.0
Voonik                                                      5000000.0
RezNext                                                     5000000.0
Urban Ladder                                                50

### For point 4 (second best sector count-wise), which company received the highest investment?

In [59]:
# For USA, "SST Inc. (Formerly ShotSpotter)" is the company which received the highest investment in "Social, Finance, Analytics, Advertising" sector

D1_SFAA_df = D1.loc[D1.main_sector == 'Social, Finance, Analytics, Advertising']

D1_SFAA_df.pivot_table(values = 'raised_amount_usd', index = 'name', aggfunc = 'sum').sort_values(by='raised_amount_usd', ascending = False)

raised_amount_usd
name                                              
SST Inc. (Formerly ShotSpotter)         67933006.0
Demandbase                              63000000.0
Intacct                                 61800000.0
NetBase Solutions                       60600000.0
Lotame                                  59700000.0
FirstRain                               58344731.0
ChoiceStream                            58300000.0
OpTier                                  55600000.0
Damballa                                55000000.0
Zoove                                   54800000.0
Xactly Corp                             54500000.0
Hightail                                54000000.0
ChaCha                                  53000000.0
INRIX                                   51100018.0
Visible Measures                        48500000.0
Univa                                   48385708.0
Vuclip                                  48100000.0
YuMe                                    47900000.0
Turn                                    45500000.0
Jaspersoft                              43500000.0
edo Interactive                         43000100.0
InstaMed                                43000000.0
Inkling Systems                         42599999.0
Laszlo Systems                          41850000.0
Metamarkets                             41000000.0
Host Analytics                          40650000.0
PivotLink                               40085068.0
4INFO                                   40000000.0
Aggregate Knowledge                     39958350.0
FiberZone Networks                      39786202.0
...                                            ...
Yieldbot                                 5000000.0
FolioMetrix                              5000000.0
Lucid Software Inc                       5000000.0
Altitude Digital                         5000000.0
LoudCloud Systems                        5000000.0
Flashpoint                               5000000.0
Fisoc                                    5000000.0
TapInfluence                             5000000.0
First30Days                              5000000.0
Celect                                   5000000.0
ThisNext                                 5000000.0
FriendFeed                               5000000.0
TrackIf                                  5000000.0
OrangeSoda                               5000000.0
InfoScout                                5000000.0
OnSwipe                                  5000000.0
Manicube                                 5000000.0
Genospace                                5000000.0
FutureAdvisor                            5000000.0
Apester                                  5000000.0
Fuse Powered Inc.                        5000000.0
FundsXpress Financial Network            5000000.0
Initiate Systems                         5000000.0
Inneractive                              5000000.0
Amtec                                    5000000.0
Bulldog Solutions                        5000000.0
Fundly                                   5000000.0
Thredhq                                  5000000.0
Xylo, Inc                                5000000.0
Convo                                    5000000.0

[1847 rows x 1 columns]

In [60]:
# For GBR, "Celltick Technologies" is the company which received the highest investment in "Social, Finance, Analytics, Advertising" sector

D2_SFAA_df = D2.loc[D2.main_sector == 'Social, Finance, Analytics, Advertising']

D2_SFAA_df.pivot_table(values = 'raised_amount_usd', index = 'name', aggfunc = 'sum').sort_values(by='raised_amount_usd', ascending = False)

raised_amount_usd
name                                                   
Celltick Technologies                        37500000.0
myThings                                     34000000.0
Zopa                                         32900000.0
VisualDNA                                    28550000.0
MarketInvoice                                25553007.0
SumUp                                        24293649.0
Amplience                                    24200000.0
Garlik                                       22350000.0
SportPursuit                                 22024487.0
Victor                                       21500000.0
OpenGamma                                    21000000.0
Kantox                                       19646334.0
Troux Technologies                           17850000.0
Neul                                         17800000.0
Livebookings Holdings Ltd.                   16500000.0
Crowdcube                                    15862857.0
Coco Communications                          15047666.0
SimilarWeb                                   15000000.0
The Exchange Lab                             14000000.0
PurpleBricks                                 13364740.0
iCrossing                                    13000000.0
Captify                                      12597433.0
Borro                                        12072758.0
Metail                                       12000000.0
Videoplaza                                   12000000.0
Inspired Capital plc                         12000000.0
VirtenSys                                    12000000.0
Seedrs                                       11703571.0
Psytechnics                                  11600000.0
Ocean Outdoor                                11597927.0
...                                                 ...
SalaryFinance                                 6100000.0
TransferWise                                  6000000.0
Peoplevox                                     6000000.0
Hassle.com                                    6000000.0
Wonga                                         6000000.0
Brandwatch                                    6000000.0
Yieldr                                        5918360.0
Tideway Systems ltd                           5800000.0
Quipper                                       5800000.0
Rockabox                                      5697713.0
Moda in Pelle                                 5613247.0
Ebury                                         5603526.0
Linkdex                                       5509959.0
aiHit                                         5500000.0
365Scores                                     5500000.0
Somo                                          5500000.0
Portr                                         5500000.0
Mirriad                                       5457478.0
VouchedFor                                    5400000.0
SCL                                           5325912.0
Nutmeg                                        5322729.0
smartfundit.com                               5170000.0
Palringo                                      5164439.0
Memrise                                       5100000.0
Government Grant and Tax Consultants          5015782.0
The Fizzback Group                            5000000.0
Tamoco                                        5000000.0
Rangespan                                     5000000.0
MOO.COM                                       5000000.0
DueDil                                        5000000.0

[104 rows x 1 columns]

In [61]:
# For IND, "Manthan Systems" is the company which received the highest investment in "Social, Finance, Analytics, Advertising" sector

D3_SFAA_df = D3.loc[D3.main_sector == 'Social, Finance, Analytics, Advertising']

D3_SFAA_df.pivot_table(values = 'raised_amount_usd', index = 'name', aggfunc = 'sum').sort_values(by='raised_amount_usd', ascending = False)

raised_amount_usd
name                                                             
Manthan Systems                                        50700000.0
Komli Media                                            28000000.0
ShopClues.com                                          25000000.0
inTarvo                                                21900000.0
Grameen Financial Services                             21556050.0
BankBazaar.com                                         19000000.0
Microland                                              18300000.0
Eka Software Solutions                                 16000000.0
Qyuki                                                  15400000.0
Financial Information Network & Operations Pvt         15000000.0
ApnaPaisa                                              15000000.0
Shiram Credit                                          15000000.0
TravelGuru                                             15000000.0
Capital Float                                          13000000.0
yepme.com                                              12700000.0
Coverfox Insurance                                     12000000.0
FundsIndia                                             11000000.0
Emergent Ventures India                                10040000.0
Network18                                              10000000.0
Excellence4u                                           10000000.0
Lendingkart                                            10000000.0
LogiNext Solutions                                     10000000.0
Unicommerce eSolutions Pvt. Ltd.                       10000000.0
Razorpay                                                9000000.0
Industrybuying                                          9000000.0
Vizury                                                  9000000.0
Fusion Microfinance                                     8300000.0
Prizm Payment Services                                  8000000.0
Shubham Housing Development Finance Company             8000000.0
Spire Technologies                                      8000000.0
Epiance                                                 7500000.0
IndiaIdeas                                              7500000.0
CRMNEXT                                                 7003500.0
Neogrowth                                               7000000.0
Minglebox                                               7000000.0
A LITTLE WORLD                                          6410000.0
Furlenco                                                6000000.0
Daksh Infosoft                                          6000000.0
Agile                                                   5740000.0
Eko India Financial Services                            5500000.0
EBS Worldwide Services                                  5500000.0
Amagi Media Labs                                        5500000.0
PolicyBazaar                                            5000000.0
Annapurna Microfinace                                   5000000.0
authorSTREAM.com                                        5000000.0
ideacts innovations                                     5000000.0
Akosha                                                  5000000.0
Innoviti                                                5000000.0